ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts

```diff
@@ -438,13 +433,55 @@ export function useHomeOwners (isMhrTransfer: boolean = false) {
 
       Object.assign(groupToUpdate, {
         ...fractionalData,
-        ...(hasChanges && { action: ActionTypes.CHANGED })
+        ...(hasChanges && { action: isMhrTransfer ? ActionTypes.CHANGED : ActionTypes.CORRECTED })
       })
 
       const updatedOwnerGroups = [...homeOwnerGroups]
       setTransferOrRegistrationHomeOwnerGroups(updatedOwnerGroups)
     }
 
+  /**
+   * Return the baseline owners snapshot by id.
+   * @param ownerId The owner identifier
+   */
+  const getMhrBaselineOwnerById = (ownerId: number): MhrRegistrationHomeOwnerIF => {
+    let currentOwner: MhrRegistrationHomeOwnerIF
+    getMhrBaseline.value?.ownerGroups.forEach(group => group.owners.find(owner => {
+      if (owner.ownerId === ownerId) currentOwner = owner
+    }))
+
+    return currentOwner
+  }
+
+  /**
+   * Return the baseline owner group snapshot by id.
+   * @param groupId The group identifier
+   */
+  const getMhrBaselineOwnerGroupById = (groupId: number): MhrRegistrationHomeOwnerGroupIF => {
+    return getMhrBaseline.value?.ownerGroups[groupId-1]
+  }
+
+  /** Return true if the specified owner has been modified from current state **/
+  const isCorrectedOwner = (owner: MhrRegistrationHomeOwnerIF): boolean => {
+    const currentOwner = getMhrBaselineOwnerById(owner.ownerId)
+    const isEqualName = !!owner.individualName
+      ? deepChangesComparison(currentOwner.individualName, owner.individualName)
+      : deepChangesComparison(currentOwner.organizationName, owner.organizationName)
+
+    return currentOwner && (
```

Comment:
This `currentOwner` check happens after it's already been used to get `isEqualName`. This could cause runtime errors. Maybe use optional chaining for `currentOwner` when comparing the names?

Reply:
This won't cause runtime errors, no. We know we have a currentOwner, in corrections. 
To this point, we probably don't need the short circuit however.

If we were dealing with some reactive properties that perhaps started empty, it would make sense to protect against i suppose.

To your point though, anything and everything could* cause runtime errors, we don't optionally chain every single accessor.

Reply:
Ok, so if you are saying the we will always have the current owner (and won't cause errors) then I'm good.

Reply:
What caused my concern is the check for `currentOwner` in the return statement.

Reply:
If we have an MHR Correction without any current owners, we have other issues you bet but your concern is noted Dima.

Reply:
Removed the currentOwner check in the return statement that was causing concern

Reply:
Thank you